autogpt/processing/text.py

```diff
@@ -12,23 +11,25 @@
 
 
 def split_text(text: str, max_length: int = 8192) -> Generator[str, None, None]:
-    """Split text into chunks of a maximum length
-
-    Args:
-        text (str): The text to split
-        max_length (int, optional): The maximum length of each chunk. Defaults to 8192.
-
-    Yields:
-        str: The next chunk of text
-
-    Raises:
-        ValueError: If the text is longer than the maximum length
-    """
     paragraphs = text.split("\n")
     current_length = 0
     current_chunk = []
 
+    def split_long_paragraph(paragraph: str, max_length: int) -> List[str]:
+        return [
+            paragraph[i : i + max_length] for i in range(0, len(paragraph), max_length)
```

Comment:
I'm not sure how much of a difference it makes for the performance of the LLM, but could you try splitting it on a whitespace (or other non-word) character instead of exactly on the `max_length`?

Reply:
after a thorough review of the split_text function I found out we can simplify it a lot using `textwrap`. No need to have this split_long_paragraph function anymore. 

Please check the new revised split_text. Also you might want to check the tests I added